code/modules/randomMaps/vaults.dm

```diff
@@ -0,0 +1,98 @@
+//Vaults are structures that are randomly spawned as a part of the main map
+//They're stored in maps/randomVaults/ as .dmm files
+
+//HOW TO ADD YOUR OWN VAULTS:
+//1. make a map in the maps/randomVaults/ folder (1 zlevel only please)
+//2. add the map's name to the vault_map_names list
+//3. the game will handle the rest
+
+#define MINIMUM_VAULT_AMOUNT 1 //Amount of guaranteed vault spawns
+
+//#define SPAWN_ALL_VAULTS //Uncomment to spawn all existing vaults (otherwise only some will spawn)!
+
+var/const/vault_map_directory = "maps/randomVaults/"
+
+
+var/list/vault_map_names = list( //Add your vaults' map names to this list. Don't include the .dmm prefix
+	"doomed_satelite",
+	"asteroid_temple",
+	"icetruck_crash",
+)
+
+/area/random_vault
+	name = "random vault area"
+	desc = "Spawn a vault in there somewhere"
+	icon_state = "random_vault"
+
+//Because areas are shit and it's easier that way!
+
+//Each of these areas can only create ONE vault. Only using /area/random_vault/v1 for the entire map will result in ONE vault being created.
+//Placing them over (or even near) shuttle docking ports will sometimes result in a vault spawning on top of a shuttle docking port. This isn't a big problem, since
+//shuttles can destroy the vaults, but it's better to avoid that
+//If you want more vaults, feel free to add more subtypes of /area/random_vault. You don't have to add these subtypes to any lists or anything - just map it and the game will handle the rest.
+
+//"/area/random_vault" DOESN'T spawn any vaults!!!
+/area/random_vault/v1
```

Comment:
Fucking area code.

THANKS BYOND.

To be honest this is probably the rare exception where I'm fine with map instances, but whatever.